### ISSUE TITLE
Disjunction when exporting files from docker.#Run if no suffix is used

### ISSUE THREAD

**Issue by TomChv** [dagger/dagger#1478]
## Bug

I found a small issue on `docker.#Run` when I tested `export` feature

Here a code snippet with the explanation
```cue
package test

import (
    "dagger.io/dagger"
    "dagger.io/dagger/engine"
    "universe.dagger.io/docker"
    "universe.dagger.io/alpine"
)

dagger.#Plan & {
    actions: {
        image: alpine.#Build

        run: docker.#Run & {
            "image": image.output
            script: #"""
                echo -n hello_world >> /output.txt
                """#
            // export: files: "/output.txt" <- DISJUNCTION
            export: files: "/output.txt": {}
        }

        verify: engine.#ReadFile & {
            input: run.output.rootfs
            path:  "/output.txt"
        } & {
            contents: run.export.files."/output.txt".contents
        }
    }
}
```

If I do `export: files: "/output.txt"` I got an ugly disjunction error

```shell
 dagger --europa up run-export-files-test.cue
4:32PM FTL failed to load plan: actions.run: 1 errors in empty disjunction:
actions.run.export.files: conflicting values "/output.txt" and {[string]:{contents:(string & _read(universe.dagger.io/docker).contents),_read(universe.dagger.io/docker):("dagger.io/dagger/engine".#ReadFile & {input:_exec(universe.dagger.io/docker).output,path:path})}} (mismatched types string and struct):
    /Users/tomchauveau/Documents/DAGGER/dagger/pkg/universe.dagger.io/docker/run.cue:89:2
    /Users/tomchauveau/Documents/DAGGER/dagger/pkg/universe.dagger.io/docker/run.cue:107:11
    /Users/tomchauveau/Documents/DAGGER/dagger/pkg/universe.dagger.io/docker/test/run-export-files-test.cue:10:1
    /Users/tomchauveau/Documents/DAGGER/dagger/pkg/universe.dagger.io/docker/test/run-export-files-test.cue:14:8
    /Users/tomchauveau/Documents/DAGGER/dagger/pkg/universe.dagger.io/docker/test/run-export-files-test.cue:19:19
```

But it works if I do `export: files: "/output.txt": {}`

We should improve the error because it's really hard to figure out the issue if you are not used to CUE disjunction

/cc @aluzzardi

### code
```
package test

import (
	"dagger.io/dagger"
	"dagger.io/dagger/core"
	"universe.dagger.io/docker"
	"universe.dagger.io/alpine"
)

dagger.#Plan & {
	actions: {
		image: alpine.#Build

		run: docker.#Run & {
			input: image.output
			command: {
				name: "echo"
				flags: {
					"-n": true
				}
				args: [ "hello_world", ">>", "/output.txt"]
			}
			// export: files: "/output.txt" <- DISJUNCTION
			export: files: "/output.txt": {}
		}

		verify: core.#ReadFile & {
			input: run.output.rootfs
			path:  "/output.txt"
		} & {
			contents: run.export.files."/output.txt".contents
		}
	}
}
```

### log

It improved a little

```
$ dagger do run
failed to load plan: actions.run.export.files: conflicting values "/output.txt" and {[string]:string} (mismatched types string and struct):
    ./cue.mod/pkg/universe.dagger.io/docker/run.cue:104:2
    ./cue.mod/pkg/universe.dagger.io/docker/run.cue:122:11
    ./repro.cue:10:1
    ./repro.cue:14:8
    ./repro.cue:23:19


```

**Comment by helderco** [comment]
Maybe the same solution from #1426 (detecting incompatible types) can help with this.

**Comment by jpadams** [comment]
Glad we have this explanation in the docs as to why you need `{}` or `_`, though:
https://docs.dagger.io/1241/docker#exports

**Comment by TomChv** [comment]
If we already have an explanation, should we keep this issue open?

**Comment by jlongtine** [comment]
I'd suggest we close it, @TomChv

**Comment by TomChv** [comment]
Ok, the issue is closed because we solved it in #613.
